app/src/main/java/com/davidtomas/taskyapp/features/agenda/data/agenda/repository/AgendaRepository.kt

```diff
@@ -18,28 +18,28 @@ class AgendaRepository(
     private val reminderLocalSource: ReminderLocalSource,
     private val taskLocalSource: TaskLocalSource
 ) {
-    suspend fun getAgenda(): Result<Flow<List<AgendaItem>>, DataError.Network> =
+    suspend fun getAgenda(): Result<Flow<List<AgendaModel>>, DataError.Network> =
         coroutineScope {
             val jobEvents = async { eventLocalSource.getEvents() }
             val jobTasks = async { taskLocalSource.getTasks() }
             val jobReminders = async { reminderLocalSource.getReminder() }
             val eventsResult = jobEvents.await()
             val tasksResult = jobTasks.await()
             val remindersResult = jobReminders.await()
-            val eventsFlow = eventsResult.getOrNull() ?: flow { listOf<AgendaItem>() }
-            val tasksFlow = tasksResult.getOrNull() ?: flow { listOf<AgendaItem>() }
-            val remindersFlow = remindersResult.getOrNull() ?: flow { listOf<AgendaItem>() }
-            val listAgendaItem = mutableListOf<AgendaItem>()
+            val eventsFlow = eventsResult.getOrNull() ?: flow { listOf<AgendaModel>() }
+            val tasksFlow = tasksResult.getOrNull() ?: flow { listOf<AgendaModel>() }
+            val remindersFlow = remindersResult.getOrNull() ?: flow { listOf<AgendaModel>() }
+            val listAgendaModel = mutableListOf<AgendaModel>()
```

Comment:
If you're using flows, you don't need to take care of these independent coroutines anymore, unless you do this inside a single flow operator. So if you want to return a stream from `getAgenda`, you wouldn't return a `Result<Flow<..>, ...>`, but rather a `Flow<Result<.., ..>>`, so each emission comes with data whether it was successful or not. Also, what could really go wrong in this function? You just load local items (and return a network error result btw), so it could even just be a `Flow<List<AgendaModel>>` and all you need to return is your combiner below